web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -253,21 +249,13 @@ private void connectToWorker() {
                     if (metadata().has(FLIGHT_AUTH_HEADER_NAME)) {
                         return authUpdate().then(ignore -> Promise.resolve(Boolean.FALSE));
                     }
-                    return Promise.all(
-                            info.getConnectToken().then(authToken -> {
-                                // set the proposed initial token and make the first call
-                                metadata.set(FLIGHT_AUTH_HEADER_NAME,
-                                        (authToken.getType() + " " + authToken.getValue()).trim());
-                                return Promise.resolve(authToken);
-                            }),
-                            info.getConnectOptions().then(options -> {
-                                // set other specified headers, if any
-                                JsObject.keys(options.headers).forEach((key, index) -> {
-                                    metadata.set(key, options.headers.get(key));
-                                    return null;
-                                });
-                                return Promise.resolve(options);
-                            })).then(ignore -> authUpdate()).then(ignore -> Promise.resolve(Boolean.TRUE));
+                    metadata.set(FLIGHT_AUTH_HEADER_NAME,
+                            (info.getToken().getType() + " " + info.getToken().getValue()).trim());
+                    JsObject.keys(info.getOptions().headers).forEach((key, index) -> {
+                        metadata.set(key, info.getOptions().headers.get(key));
+                        return null;
+                    });
+                    return authUpdate().then(ignore -> Promise.resolve(Boolean.TRUE));
```

Comment:
Do we have to use `Promise.resolve` in the return? I thought non-promise and promise returns from a `then` both result in the same thing.

Reply:
In JS yes, but in Java you cannot define a method signature to return two unrelated types - union types effectively don't exist in this fashion (and while _technically_ jvm bytecode allows method signatures to be specified based on return type, the java language doesn't have support for that). So, when we want to return a value directly, we must always wrap in Promise.resolve(...).